File: CS/XPServerModeViewDemo/DataAccess/Order.cs
using DevExpress.Xpo;
using System;

namespace XpoTutorial {

    public class Order :XPObject {
        public Order(Session session) : base(session) { }
        public string ProductName {
            get { return GetPropertyValue<string>(nameof(ProductName)); }
            set { SetPropertyValue(nameof(ProductName), value); }
        }
        public DateTime OrderDate {
            get { return GetPropertyValue<DateTime>(nameof(OrderDate)); }
            set { SetPropertyValue(nameof(OrderDate), value); }
        }
        public decimal Price {
            get { return GetPropertyValue<decimal>(nameof(Price)); }
            set { SetPropertyValue(nameof(Price), value); }
        }
        public decimal Quantity {
            get { return GetPropertyValue<decimal>(nameof(Quantity)); }
            set { SetPropertyValue(nameof(Quantity), value); }
        }
        [Association("CustomerOrders")]
        public Customer Customer {
            get { return GetPropertyValue<Customer>(nameof(Customer)); }
            set { SetPropertyValue(nameof(Customer), value); }
        }
    }
}

File: VB/XPServerModeViewDemo/DataAccess/Order.vb
Imports DevExpress.Xpo
Imports System

Namespace XpoTutorial

	Public Class Order
		Inherits XPObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property ProductName() As String
			Get
				Return GetPropertyValue(Of String)(NameOf(ProductName))
			End Get
			Set(ByVal value As String)
				SetPropertyValue(NameOf(ProductName), value)
			End Set
		End Property
		Public Property OrderDate() As Date
			Get
				Return GetPropertyValue(Of Date)(NameOf(OrderDate))
			End Get
			Set(ByVal value As Date)
				SetPropertyValue(NameOf(OrderDate), value)
			End Set
		End Property
		Public Property Price() As Decimal
			Get
				Return GetPropertyValue(Of Decimal)(NameOf(Price))
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue(NameOf(Price), value)
			End Set
		End Property
		Public Property Quantity() As Decimal
			Get
				Return GetPropertyValue(Of Decimal)(NameOf(Quantity))
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue(NameOf(Quantity), value)
			End Set
		End Property
		<Association("CustomerOrders")>
		Public Property Customer() As Customer
			Get
				Return GetPropertyValue(Of Customer)(NameOf(Customer))
			End Get
			Set(ByVal value As Customer)
				SetPropertyValue(NameOf(Customer), value)
			End Set
		End Property
	End Class
End Namespace
